test/e2e2/cache_test.go

```diff
@@ -206,33 +211,39 @@ func createApp(ctx context.Context, client fixture.KubeClient, requires *require
 	return app
 }
 
-func validateManagedAppCreated(ctx context.Context, mClient fixture.KubeClient, pClient fixture.KubeClient, key types.NamespacedName, requires *require.Assertions) argoapp.Application {
+func validateManagedAppCreated(ctx context.Context, mClient fixture.KubeClient, pClient fixture.KubeClient, principalKey types.NamespacedName, agentKey types.NamespacedName, requires *require.Assertions) argoapp.Application {
 	app := argoapp.Application{}
-	requires.NoError(pClient.Get(ctx, key, &app, metav1.GetOptions{}))
+	requires.NoError(pClient.Get(ctx, principalKey, &app, metav1.GetOptions{}))
 	// The destination on the agent will be set to "in-cluster"
 	app.Spec.Destination.Name = "in-cluster"
 	app.Spec.Destination.Server = ""
 
 	// Check that the spec field of agent app matches with principal app
 	mapp := argoapp.Application{}
 	requires.Eventually(func() bool {
-		err := mClient.Get(ctx, key, &mapp, metav1.GetOptions{})
+		err := mClient.Get(ctx, agentKey, &mapp, metav1.GetOptions{})
 		return err == nil && reflect.DeepEqual(&app.Spec, &mapp.Spec)
 	}, 90*time.Second, 2*time.Second)
 
 	return app
 }
 
 func updateAppInfo(ctx context.Context, client fixture.KubeClient, key types.NamespacedName, info []string, requires *require.Assertions) {
-	requires.NoError(client.EnsureApplicationUpdate(ctx, key, func(app *argoapp.Application) error {
-		app.Spec.Info = []argoapp.Info{
-			{
-				Name:  info[0],
-				Value: info[1],
-			},
+	requires.Eventually(func() bool {
+		err := client.EnsureApplicationUpdate(ctx, key, func(app *argoapp.Application) error {
```

Comment:
It seems like the `requires.Eventually()`-style logic is already in `EnsureApplicationUpdate`? Unless there is some other case this is handling.